### PR TITLE
create script to extract layer containing colliding tiles

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

File: scripts/extractMap.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const fs = require("fs");
```
Comment (przenz) [inline scripts/extractMap.js:1]: Please add some brief description what this script does.

File: package.json [hunk 2]
```diff
@@ -41,10 +41,13 @@
     "watcherGlob": "**"
   },
   "scripts": {
+    "prebuild": "yarn extract:map",
     "build": "if test \"$NODE_ENV\" = \"production\"; then yarn build:prod; else yarn build:dev; fi",
     "build:prod": "NODE_ENV=production parcel build src/index.html",
     "build:dev": "NODE_ENV=development parcel build src/index.html --no-minify",
+    "predev": "yarn extract:map",
     "dev": "parcel watch src/index.html",
+    "extract:map": "node ./scripts/extractMap.js",
```
Comment (przenz) [inline package.json:50]: I'm not sure how this script does extract anything, maybe it should be renamed to "export" ?

File: scripts/extractMap.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  arr.push([]);
+  for (let k = 0; k < layerCollides.height; k += 1) {
+    const tileId = layerCollides.data[m];
+    arr[j][k] = tileId === 0 ? -1 : tileId;
+    m += 1;
+  }
+}
+
+fs.writeFile(
+  "./public/assets/map/map.js",
```
Comment (przenz) [inline scripts/extractMap.js:21]: Maybe it should be exported to a separate file?

**Review by przenz — CHANGES_REQUESTED** [review]

File: public/assets/map/map.js [hunk 4]
```diff
@@ -0,0 +1 @@
...,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1]];
```
[the remark this answers is not in the document]
Comment (neu5) [reply public/assets/map/map.js:1]: Right, I'll remove the file

File: package.json (hunk 2 above)
Comment (neu5) [reply package.json:50]: I feel like it extracts colliding layer, creates two dimensional array and exports it. So I'm not sure how to name it properly

File: scripts/extractMap.js (hunk 3 above)
Comment (neu5) [reply scripts/extractMap.js:21]: Yup, maybe. ATM all files from ./public are copied to ./dist. We could clean it up a bit.

Comment (przenz) [reply scripts/extractMap.js:29]: A separate issue for this would be nice

**Review by przenz — APPROVED** [review]

File: scripts/extractMap.js (hunk 3 above)
Comment (neu5) [reply scripts/extractMap.js:29]: Here you go - #59

**Pull request merged**